pkg/cmd/secrets_notes.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright © 2020 Doppler <support@doppler.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"github.com/DopplerHQ/cli/pkg/configuration"
+	"github.com/DopplerHQ/cli/pkg/http"
+	"github.com/DopplerHQ/cli/pkg/printer"
+	"github.com/DopplerHQ/cli/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var secretsNotesCmd = &cobra.Command{
+	Use:   "notes",
+	Short: "Manage secret notes",
+	Args:  cobra.NoArgs,
+}
+
+var secretsNotesSetCmd = &cobra.Command{
+	Use:   "set [secret] [note]",
+	Short: "Set a note on a secret. The secret must exist.",
+	Args:  cobra.ExactArgs(2),
+	Run:   setSecretNote,
+}
+
+func setSecretNote(cmd *cobra.Command, args []string) {
+	jsonFlag := utils.OutputJSON
+	localConfig := configuration.LocalConfig(cmd)
+
+	utils.RequireValue("token", localConfig.Token.Value)
+
+	secret := args[0]
+	note := args[1]
+
+	utils.RequireValue("secret", secret)
+
+	response, httpErr := http.SetSecretNote(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, secret, note)
+	if !httpErr.IsNil() {
+		utils.HandleError(httpErr.Unwrap(), httpErr.Message)
+	}
+
+	if !utils.Silent {
+		printer.SecretNote(response, jsonFlag)
+	}
+}
+
+func init() {
+	secretsNotesSetCmd.Flags().StringP("project", "p", "", "project (e.g. backend)")
+	secretsNotesSetCmd.Flags().StringP("config", "c", "", "config (e.g. dev)")
+	secretsNotesCmd.AddCommand(secretsNotesSetCmd)
+
+	secretsCmd.AddCommand(secretsNotesCmd)
```

Comment:
2 questions:

- Do we want the notes command to live under the secrets command. Notes are tied to secrets but the command feels kind of off:
`doppler secrets notes set MY_SECRET "Hello, World"`
- Should I be able to make a multiline note from the CLI? Adding `\n` in my note did not translate into a new line.

Reply:
- I couldn't think of a better location for it, but open to suggestions. I don't think it should live at the root `doppler notes`
- Indeed you should. I'll add support for reading the secret note from stdin, which will allow for multiline values.

Reply:
Added multiline note support https://github.com/DopplerHQ/cli/compare/11a1bed0026e6404652a66a07fbd33c410367cf4..d0ae528280d509092c15d5e2e47b9d69644e0dec

Reply:
I don't have a better suggestion of where the command should live. Like you wrote, the root isn't really where it should be either. Let's go with this for now since there's nowhere else to really put it.